### PR TITLE
Add docs

### LINKED ISSUE #52: Write docs

### LINKED ISSUE #78: Convert derived state to camelCase

### PR DESCRIPTION
**Pull request by beerinho**
This PR adds the preliminary documentation for the plugin. I found it a bit tricky to write the introduction without it sounding super cheesy so feel free to give any suggestions for improvements there

Closes #52 
Closes #78

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,175 @@
...
+Then put it to work immediately wherever you want cancellable promises.
+
+## Task structure
+
+All tasks will return a store with the same structure:
+
+- error: Error - if an error occurred, it will be returned here,
+- results: Array - all of the results from previous invocations of this task,
+- last_successful: Any - the return value from the last successful run of the task
+- is_loading: Boolean - whether the task is currently running on not
```
Comment (paoloricciuti) [inline README.md:30]: I wonder if we should change the public api to be camelCase but for the meantime is better if the docs reflects the code. Also we should discuss what other derived state we want to add.

File: README.md [hunk 2]
```diff
@@ -1,14 +1,175 @@
...
+
+### Standard/default task
+
+This simply gives you a task wrapper around your function. It will not handle any kind of concurrency for you.
+
+```ts
+<script>
+	import { task } from 'svelte-concurrency';
+
+	const my_task = task(async function* (param: number) {
```
Comment (paoloricciuti) [inline README.md:46]: For the docs i think we should probably start with async Promises in the examples and talk about generators after having introduced the concept of mid run cancelability. If we want to use async generator this is still missing a `yield` after the await 😄

File: README.md [hunk 3]
```diff
@@ -1,14 +1,175 @@
...
+		return param * 2;
+	}, { kind: 'enqueue', max: 3 });
+</script>
+```
+
+Both of the above will result in 3 simultaneous tasks being allowed to run. Any additional instances of the task will be added to a list and run when there is space in the queue.
+
+### Accessing the task in the template
+
+As the return value from the task wrapper is a store, you can access it just like you would with any other store:
```
Comment (paoloricciuti) [inline README.md:158]: We should probably mention that if you return something from the task you can await the `task.perform` to get the result (but also that if you don't await it the return value of `task.perform` is the same store.

**Comment by beerinho** [comment]
We need to add async, generators, async transform in that order.
We also need to add the options that are available in the task

**Review by paoloricciuti — APPROVED** [review]
@beerinho i took the liberty of some small tweak to the code blocks but good for me (i just committed since it was easier to just Cmd+F the change than to write the comments 😄 )

**Comment by beerinho** [comment]
You're really gonna do me like that. It was bad enough having all of the code in American English, now you're trying to make the docs the same? 🤢

**Pull request merged**